core/es_isolate.rs

```diff
@@ -1023,4 +1025,32 @@ pub mod tests {
       assert_eq!(load_count.load(Ordering::Relaxed), 2);
     })
   }
+
+  #[test]
+  fn dyn_import_borrow_mut_error() {
+    // https://github.com/denoland/deno/issues/6054
+    run_in_task(|cx| {
+      let loader = Rc::new(DynImportOkLoader::default());
+      let prepare_load_count = loader.prepare_load_count.clone();
+      let mut isolate = EsIsolate::new(loader, StartupData::None, false);
+      js_check(isolate.execute(
+        "file:///dyn_import3.js",
+        r#"
+          (async () => {
+            let mod = await import("./b.js");
+            if (mod.b() !== 'b') {
+              throw Error("bad");
+            }
+            // Now do any op
+            Deno.core.ops();
+          })();
```

Comment:
Nitpick: not needed to wrap in IIFE

Reply:
actually it is necessary - the tests aren't configured to allow top-level-await apparently